src/inngest/functions/create-course/utils/upsertCourseToTypesense.ts

```diff
@@ -0,0 +1,76 @@
+import Typesense from 'typesense'
+import z from 'zod'
+import {syncWithSearchProvider} from '@/lib/search'
+
+let client = new Typesense.Client({
+  nodes: [
+    {
+      host: process.env.NEXT_PUBLIC_TYPESENSE_HOST!,
+      port: 443,
+      protocol: 'https',
+    },
+  ],
+  apiKey: process.env.TYPESENSE_WRITE_API_KEY!,
+  connectionTimeoutSeconds: 2,
+})
+
+const courseSchema = z.object({
+  id: z.number(),
+  slug: z.string(),
+  state: z.string(),
+  visibility_state: z.string(),
+  summary: z.string(),
+  free_forever: z.boolean(),
+  type: z.literal('playlist'),
+  created_at: z.string(),
+  published: z.boolean(),
+  rating_out_of_5: z.number(),
+  square_cover_url: z.string(),
+  topic_list: z.array(z.string()),
+  updated_at: z.string(),
+  instructor: z.object({
+    id: z.number(),
+    full_name: z.string(),
+  }),
+  tags: z.array(
+    z.object({
+      name: z.string(),
+      slug: z.string(),
+      label: z.string(),
+      image_url: z.string(),
+      http_url: z.string(),
+      description: z.string(),
+    }),
+  ),
+  title: z.string(),
+  description: z.string(),
+  path: z.string(),
+})
```

Comment:
one thing I notice is that every property in this schema is required

is that the actual case? if not can probably loosen the type up a bit.